### PR TITLE
executor: fix the wrong `order by pk desc` result and some corner cases for unsigned pk

### LINKED ISSUE #10178: Max result incorrect when using Unsigned bigint as primary key.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
create table test_uint64(c1 bigint unsigned, primary key(c1));
insert into test_uint64 values(9223372036854775807);
insert into test_uint64 values(18446744073709551615);
select max(c1) from test_uint64;
```

2. What did you expect to see?
```
mysql> select max(c1) from test_uint64;
+----------------------+
| max(c1)              |
+----------------------+
| 18446744073709551615 |
+----------------------+
1 row in set (0.01 sec)
```

3. What did you see instead?
```
mysql> select max(c1) from test_uint64;
+---------------------+
| max(c1)             |
+---------------------+
| 9223372036854775807 |
+---------------------+
1 row in set (0.01 sec)
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

Any version will do.

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

fix #10178 

### What is changed and how it works?

When split ranges, we forget to check the `DESC` status and some corner case checking.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/distsql_test.go [hunk 1]
```diff
@@ -172,3 +172,15 @@ func (s *testSuite3) TestUniqueKeyNullValueSelect(c *C) {
...
+// TestIssue10178 contains tests for https://github.com/pingcap/tidb/issues/10178 .
+func (s *testSuite3) TestIssue10178(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a bigint unsigned primary key)")
+	tk.MustExec("insert into t values(9223372036854775807), (18446744073709551615)")
+	tk.MustQuery("select max(a) from t").Check(testkit.Rows("18446744073709551615"))
+	tk.MustQuery("select * from t where a > 9223372036854775807").Check(testkit.Rows("18446744073709551615"))
+	tk.MustQuery("select * from t where a < 9223372036854775808").Check(testkit.Rows("9223372036854775807"))
```
Comment (qw4990) [inline executor/distsql_test.go]: It seems `select * from t where a > 9223372036854775807-0.5` has a wrong result..
```
+----------------------+
| a                    |
+----------------------+
| 18446744073709551615 |
+----------------------+
```

Comment (winoros) [reply executor/distsql_test.go]: It should be another separate problem.
Before split ranges, it's already become `(9223372036854775808,+inf]`.

File: executor/distsql.go [hunk 2]
```diff
@@ -174,27 +174,37 @@ func splitRanges(ranges []*ranger.Range, keepOrder bool) ([]*ranger.Range, []*ra
...
-		LowVal:     ranges[idx].LowVal,
-		LowExclude: ranges[idx].LowExclude,
-		HighVal:    []types.Datum{types.NewUintDatum(math.MaxInt64)},
-	})
-	unsignedRanges = append(unsignedRanges, &ranger.Range{
-		LowVal:      []types.Datum{types.NewUintDatum(math.MaxInt64 + 1)},
-		HighVal:     ranges[idx].HighVal,
-		HighExclude: ranges[idx].HighExclude,
-	})
+	if !(ranges[idx].LowVal[0].GetUint64() == math.MaxInt64 && ranges[idx].LowExclude) {
```
Comment (XuHuaiyu) [inline executor/distsql.go]: Why do we need this check?

Comment (winoros) [reply executor/distsql.go]: Because there can be cases like `(maxInt64, maxInt64]` and [maxInt64+1, maxInt64+1)`. And then error thrown when send requests.

**Comment by shenli** [comment]
Please cherry-pick this to the release-2.1 branch.
